site/docs/references/custom-selectors.md

```diff
@@ -0,0 +1,164 @@
+---
+id: references/custom-selectors
+title: Custom Selectors
+layout: docs
+---
+
+You use [Custom selectors](https://drafts.csswg.org/css-extensions/#custom-selectors) to define a local name alias for complex selectors.
+
+## Usage
+
+The following code example demonstrate a `custom-selector` alias named `controlBtn` that matches any `.btn` CSS class nested under the `.controls` CSS class:
+
+**CSS API**
+```css
+@namespace "Comp";
+@custom-selector :--controlBtn .controls .btn;
+/*
+selector: .Comp__root .Comp__controls .Comp__btn
+*/
+:--controlBtn { border: 1px solid grey; }
+/*
+selector: .Comp__root .Comp__controls .Comp__btn:hover
+*/
+:--controlBtn:hover { border-color: red; }
+```
+
+### Expose pseudo-element
+
+Custom selectors generate a [pseudo-element](./pseudo-elements.md), So for example [importing](./imports.md) the previous stylesheet into another stylesheet allows access to the `form-input` pseudo-element:
+
+```css
+@namespace "Page";
+:import {
+    -st-from: "./comp.st.css";
+    -st-default: Comp;
+}
+/*
+selector: .Page__root .Comp__root .Comp__controls .Comp__btn
+*/
+Comp::controlBtn { 
```

Comment:
Should we adhere to the (proto-)spec and use the empty vendor prefix at the beginning of a custom selectorl, eg Comp::--controlBtn {color: goldenrod;} ?

I suggest we should, to keep it closer to spec.

Reply:
Where in the spec does custom selectors used as pseudo-elements? I like to take the base name and not the full custom pseudo-element like syntax when referring from outside the component. I think the fact that a component defined some pseudo element from a custom selector should be transparent to developers using a component

Reply:
Yup, fair enough.